src/routeGeneration/templateHelpers.ts

```diff
@@ -383,16 +394,27 @@ export class ValidationService {
       } else if (additionalProperties === false) {
         Object.keys(value).forEach((key: string) => {
           if (!alreadyValidatedProperties.has(key)) {
-            fieldErrors[parent + '.' + key] = {
-              message: `${key} is an excess property and therefore is not allowed`,
-              value: key,
-            };
+            if (swaggerConfig.noImplicitAdditionalProperties === 'throw-on-extras') {
+              fieldErrors[parent + '.' + key] = {
```

Comment:
```suggestion
              fieldErrors[parent + key] = {
```

Reply:
@WoH I can’t change that since it would be a breaking contract for consumers. I actually have my own custom TSOA error parser and it depends on the specific format.

Reply:
Just means the next release should be a major version.
This should definitely be fixed IMHO.

Reply:
@WoH if you think that this `parent + ‘.’ key` syntax is an issue, please open up a new PR explaining why it’s a bug. It has not been made clear why this is a bug. Additionally it’s been a part of tsoa for a long time.